src/d3d9/d3d9_swapchain.cpp

```diff
@@ -1160,4 +1170,13 @@ namespace dxvk {
     return this->GetParent()->IsExtended() ? "D3D9Ex" : "D3D9";
   }
 
+  void D3D9SwapChainEx::ValidateBackBuffersPtrs() {
+    for (uint32_t i = 0; i < m_backBuffers.size(); i++) {
+      if (!GetBackBuffer(i))
+      {
+        throw DxvkError("D3D9SwapChainEx: Back buffer pointer is invalid - nullptr");
+      }
+    }
+  }
+
```

Comment:
This shouldn't be possible to hit unless the app ignores the Reset error and goes on trying to present black frame buffers...

If you want to sanity check, just check `m_backBuffers[0]` being non-null at present time and log+error, all of them don't need to be checked.

Reply:
> This shouldn't be possible to hit unless the app ignores the Reset error and goes on trying to present black frame buffers...

Yes, agree, that makes sense, but apparently MSFT Multisample Stability (WoW64) test behaves this way and goes with Present despite of OOM error on Reset.

> If you want to sanity check, just check `m_backBuffers[0]` being non-null at present time and log+error, all of them don't need to be checked.

Actually, in my scenario, only m_backBuffers[1] is null, m_backBuffers[0] is fine, so unless we're okay with crash on Present called after failed Reset, we should validate both backbuffers.

Reply:
Any progress with this?